src/random_ray/flat_source_domain.cpp

```diff
@@ -247,12 +252,39 @@ double FlatSourceDomain::compute_k_eff(double k_eff_old) const
       sr_fission_source_new += nu_sigma_f * scalar_flux_new_[idx];
     }
 
+    // Temporary FSR fission rate for performance.
+    double temp_sr_fission_rate = sr_fission_source_new * volume;
+
+    // Calculating old and new fission rates.
     fission_rate_old += sr_fission_source_old * volume;
-    fission_rate_new += sr_fission_source_new * volume;
+    fission_rate_new += temp_sr_fission_rate;
+
+    // Assigning the fission source to the vector for entropy calculation.
+    p[sr] = temp_sr_fission_rate;
```

Comment:
Might be simpler to cut the temporary variable and just do something like
```C++
// Compute total fission rates in FSR
sr_fission_source_old *= volume;
sr_fission_source_new *= volume;

// Accumulate totals
fission_rate_old += sr_fission_source_old;
fission_rate_new += sr_fission_source_new;

// Store total fission rate in the FSR for Shannon calculation
p[sr] = sr_fission_source_new;

```

Reply:
I like that approach, thanks for the suggestion!